### PR TITLE
fix: add types

### PR DESCRIPTION
**Pull request by abdulsattar**
Add Typescript definitions to the API.

### CODE REVIEW

**Comment by rui-rayqiu** [comment]
LGTM. I'm just curious how we can test this type def somehow?

**Review by rui-rayqiu — APPROVED** [review]

**Comment by kevinv11n** [comment]
Have you considered migrating the project to typescript and generating the types export from the source?

**Comment by abdulsattar** [comment]
@kevinv11n done.

File: src/utils.ts [hunk 1]
```diff
@@ -6,44 +6,44 @@
  */
 // Provides temporary backward compatibility for wire-protocol reform (lwc > 1.5.0). This code
 // should be removed once all adapters are migrated to the the API.
-const wireAdaptersRegistryHack = global.wireAdaptersRegistryHack || new Map();
+const wireAdaptersRegistryHack = (globalThis as any).wireAdaptersRegistryHack || new Map();
```
Comment (abdulsattar) [inline src/utils.ts:9]: This file is deprecated, so, I didn't add any types for this.

File: src/index.ts [hunk 2]
```diff
@@ -4,32 +4,33 @@
...
+    'https://github.com/salesforce/wire-service-jest-util/blob/master/docs/migrating-from-version-2.x-to-3.x.md';
 const knownAdapterMocks = new WeakSet();
 
-function getMigrationMessageFor(registerFnName) {
+function getMigrationMessageFor(registerFnName: string) {
     return `${registerFnName} is deprecated. More details: ${MIGRATION_LINK}`;
 }
 
-function validateAdapterId(adapterId) {
+function validateAdapterId(adapterId: any) {
```
Comment (abdulsattar) [inline src/index.ts:20]: These functions are deprecated, so I just put any everywhere.

File: src/__tests__/index.test.ts [hunk 3]
```diff
@@ -4,34 +4,36 @@
...
-        const testWireAdapter = target.registerLdsTestWireAdapter(()=>{});
+        const testWireAdapter = target.registerLdsTestWireAdapter(() => {});
         expect(testWireAdapter).toHaveProperty('emit');
         expect(testWireAdapter).toHaveProperty('error');
         expect(testWireAdapter).toHaveProperty('getLastConfig');
     });
 
     it('throws error when no adapter id', () => {
         expect(() => {
+            // @ts-ignore
```
Comment (rui-rayqiu) [inline src/__tests__/index.test.ts:19]: Is this because the function is deprecated?

File: src/adapters/TestWireAdapter.ts [hunk 4]
```diff
@@ -4,11 +4,18 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
+
+export interface HttpFetchResponse {
```
Comment (rui-rayqiu) [inline src/adapters/TestWireAdapter.ts:8]: In README there is a type/object called FetchResponse that looks bit different than here. Is it the same type or maybe we need to update README?

**Comment by rui-rayqiu** [comment]
LGTM overall with only minor comments. Do you think we need to update README to include an example in TS? Also I'm curious on how did you test this on core or as a consumer of the package?

**Pull request merged**